src/window.rs

```diff
@@ -0,0 +1,177 @@
+use core::marker::PhantomData;
+use core::f64;
+
+use {Sample, FloatSample};
+use signal;
+use signal::{Phase, ConstHz};
+use conv::{FromSample, ToSample};
+use frame::Frame;
+
+// Using types instead of enum allows for static dispatch
+pub trait Type<S: Sample> {
+    fn at_phase(phase: S) -> S;
+}
+
+pub struct Hanning;
+
+pub struct Rectangle;
+
+pub struct Window<F, WT> 
+    where F: Frame,
+          WT: Type<F::Sample>
+{
+    pub phase: Phase<ConstHz>,
+    ftype: PhantomData<F>,
+    wttype: PhantomData<WT>
+}
+
+pub struct WindowedFrame<'a, F, WT> 
+    where F: 'a + Frame,
+          F::Sample: FromSample<f64>, 
+          WT: Type<F::Sample>
+{
+    pub data: &'a [F],
+    pub window: Window<F, WT>,
+    idx: usize
+}
+
+/// Windower takes a long slice of data and generates an iterator over its frames
+pub struct Windower<'a, F, WT> 
+    where F: 'a + Frame, 
+          F::Sample: FromSample<f64>, 
+          WT: Type<F::Sample>
+{
+    pub bin: usize,
+    pub hop: usize,
+    pub idx: usize,
+    pub data: &'a [F],
+    wttype: PhantomData<WT>
+}
+
+impl<S: Sample + ToSample<f64> + FromSample<f64>> Type<S> for Hanning {
+    fn at_phase(phase: S) -> S {
+        let v = phase.to_sample::<f64>() * f64::consts::PI * 2.;
+        (0.5f64 * (1f64 - super::cos(v))).to_sample::<S>()
+    }
+}
+
+impl<S: Sample + FromSample<f64>> Type<S> for Rectangle {
+    #[allow(unused)]
+    fn at_phase(phase: S) -> S {
```

Comment:
When you have a binding to an unused value, you can silence the "unused" warning by adding an underscore to the beginning of the name, i.e. `fn at_phase(_phase: S) -> S`. This way you don't need the `#[allow(unused)]` decorator.